src/main/java/teammates/common/util/TimeHelper.java

```diff
@@ -215,20 +215,21 @@ public static String formatTime12H(Date date) {
         return sdf.format(date);
     }
 
-    public static String formatDateTimeForComments(Date date) {
+    public static String formatDateTimeForComments(Date date, double sessionTimeZone) {
         if (date == null) {
             return "";
         }
         SimpleDateFormat sdf = null;
         Calendar c = Calendar.getInstance(SystemParams.TIME_ZONE);
         c.setTime(date);
+        String timeZone = getCustomIdOfTimeZone(sessionTimeZone);
         if (c.get(Calendar.HOUR_OF_DAY) == 12 && c.get(Calendar.MINUTE) == 0) {
```

Comment:
You may need to update this line as well. While the calendar hour may be 12, the formatted hour may not be 12, which makes the added `" NOON"` string unsuitable. On this note, you should add more tests to TimeHelperTest to ensure your method is working correctly.